### ISSUE TITLE
Regexp group match not working

### ISSUE THREAD

**Issue by sancag** [artilleryio/artillery#2125]
Version info: 
```
Artillery: 2.0.0-36
Node.js:   v20.3.0
OS:        darwin
```

Running this command: 
```
capture:
          - regexp: '(cart#)([A-Z0-9]+)'
            group: 2
            as: 'capturedSegmentAfterHash'
```

I expected to see this happen:

With the following captured input

```
{"expiresAt":1694239938,"sk":"meta","pk":"cart#01H9W886XP9NXFPCXP57KQ4B48","isRefundableBookingSelected":false,"refundableAmount":0}}
```

I epect just the id `01H9W886XP9NXFPCXP57KQ4B48`

Instead, this happened:
I get `cart#01H9W886XP9NXFPCXP57KQ4B48` including `cart#`
```

capturedSegmentAfterHash: {
    value: 'cart#01H9W9A1N0H4SHJMSER4WZSRZ0',
    strict: undefined,
    failed: false
  }
```

**Comment by sancag** [comment]
You are a ⭐️ @InesNi . Works perfectly

**Comment by InesNi** [comment]
Glad I could help @sancag 🙌🏻

**Comment by bernardobridge** [comment]
A fix for this is now released on [Artillery v2.0.0-38](https://github.com/artilleryio/artillery/releases/tag/v2.0.0-38).
